src/WebXRPolyfill.js

```diff
@@ -127,4 +118,34 @@ export default class WebXRPolyfill {
       configurable: true,
     });
   }
+
+  _injectCompatibilityShims(global) {
+    if (!partials.every(iface => !!global[iface])) {
+      throw new Error(`Global must have the following attributes : ${partials}`);
+    }
+
+    // Patch for Chrome 76: Requires that inline XRWebGLLayers be created with
+    // { compositionDisabled: true }, so intercept the layer constructor and
+    // force it to always set that based on the session mode.
+    if (global.XRWebGLLayer) {
+      let originalRequestSession = global.navigator.xr.requestSession;
+      global.navigator.xr.requestSession = function(mode, options) {
+        return originalRequestSession.call(this, mode, options).then((session) => {
+          session._session_mode = mode;
+          return session;
+        });
+      }
+
+      var originalXRLayer = global.XRWebGLLayer;
+      global.XRWebGLLayer = function(session, gl, options) {
+        if (!options) {
+          options = {};
+        }
+
+        options.compositionDisabled = (session._session_mode === "inline");
+
+        return new originalXRLayer(session, gl, options);
```

Comment:
This actually patches a native implementation if one is available, so it can't live on the polyfilled XRWebGLLayer.

Reply:
Ahh, in that case, would it be possible for the new `global.XRWebGLLayer` to be a constructor, proxying methods to `originalXRLayer`? This would preserve the connection between the exposed constructor and the returned instance. Granted that's less important than all these other changes, maybe something to file an issue for in the future

Reply:
That's actually exactly what's happening here! Javascript conventions don't make it easy to read, but this is replacing the original constructor with a new one that returns an instance of the original XRLayer object type. Since the only thing that needs to change is the value that gets passed to the constructor, we don't need to proxy any other methods on the layer.

Reply:
For a completely contrived example, IIUC I believe the below would throw an error everytime the function is called:

```js
const layers = { webgl: new XRWebGLLayer(...), foo: new XRFooLayer(...) };
const activeLayer = layers.webgl;

function onLayerChange() {
  if (activeLayer instanceof XRWebGLLayer) {
    activeLayer = layers.foo;
  } else if (activeLayer instanceof XRFooLayer) {
    activeLayer = layers.webgl;
  } else {
    throw new Error('activeLayer is neither an instance of XRWebGLLayer or XRFooLayer');
  }
}
```

If true, it's an edge-edge-edge case, should just be something to note or issue wherever the polyfill deviates from a native implementation